### ISSUE TITLE
Pager Duty Source Plugin - Source spec config support for Team

### ISSUE THREAD

**Issue by rafaelrodrigues3092** [cloudquery/cloudquery#6981]
### Is your feature request related to a problem? Please describe.
When using the Pager Duty plugin I have encountered several API Throttling errors.
I was unable to get all the data I needed.

### Describe the solution you'd like
Introduce an optional **team** key in the source spec file config (https://www.cloudquery.io/docs/plugins/sources/pagerduty/overview).
This key would be used in the API calls to only retrieve records for this given team or teams:
This may help with:
- Large organizations where users may only want to know about incidents for certain teams
- API throttling, by potentially reducing the scope of the data pull

### Describe alternatives you've considered
I tried tunning the data pull as described here https://www.cloudquery.io/docs/advanced-topics/performance-tuning (including the experimental round-robin feature)

### Additional context
None

**Comment by shimonp21** [comment]
@rafaelrodrigues3092 Hello, and thanks for reporting this issue!
I have created a PR (#7206) to add the `team_ids` option to the Spec. A few points I can think of in the meantime:

- The most common culprits for rate-limiting are the relational tables usually (because they require an API call **per resource**). A (not-ideal) workaround can be to skip them in your config:
  ```yaml
  skip_tables: [
  "pagerduty_incident_alerts",
  "pagerduty_incident_notes",
  "pagerduty_incident_log_entries",
  "pagerduty_ruleset_rules",
  "pagerduty_user_contact_methods",
  "pagerduty_user_notification_rules"]
  ```
  
- I believe the more robust solution here would be to add a rate-limiting mechanism in the plugin itself, so we don't exceed the ~1000 requests/minute allowed by pagerduty (https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTUz-rate-limiting#what-are-our-limits). This will of course mean slower syncs, but I believe that's better than being rate limited and not bringing the data at all.

**Comment by rafaelrodrigues3092** [comment]
@shimonp21 
Thanks for this and for the pointers.
I will give it another try as soon as the new version is out.

**Comment by rafaelrodrigues3092** [comment]
> @rafaelrodrigues3092 Hi, Released version `v1.3.0`, which has both fixes:
> 
> * `team_ids` filter in the Spec,
> * Rate limiter in the plugin (only 10 API calls per second due to [PagerDuty rate limiting](https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTUz-rate-limiting#what-are-our-limits). The rate limiter is configurable via the `max_requests_per_second` option.
> 
> Note that, with the new rate-limiting, you shouldn't see rate limit errors, but the sync will be considerably slower (took 5 minutes on my toy account with 1000 incidents). Also note that the PagerDuty rate limit is **per organization**, so you may need to reduce `max_requests_per_second` farther to avoid locking up other processes in your organization.
> 
> In summary, I believe that some combination of:
> 
> 1. Skipping relational tables (since they require API-calls per resource).
> 2. Limiting `team_ids` to the relevant teams.
> 3. Using the new rate limited `v1.3.0`.
> 
> will allow you to grab the data. Do expect the sync to take longer (especially if fetching relational tables).
> 
> See configuration docs here: https://www.cloudquery.io/docs/plugins/sources/pagerduty/configuration

Many thanks @shimonp21 
I should be able to give it a go by the end of the day today!

**Comment by rafaelrodrigues3092** [comment]
@shimonp21 Both a colleague and I were able to test. This works great. Thank you!!!
